src/eei.cpp

```diff
@@ -432,7 +432,11 @@ namespace HeraVM {
 
       HERA_DEBUG << (import->base == Name("revert") ? "revert " : "return ") << hex << offset << " " << size << dec << "\n";
 
+      //zero-fill the return buffer
       result.returnValue.clear();
+      for (uint32_t i = 0; i < size; i++)
```

Comment:
This should go until `offset`, but according to @chfast this is broken anyway, we shouldn't clear and just randomly write, but rather properly `resize` which will do the zero filling.

Reply:
Apparently `resize` doesn't do zero filling if the buffer is truncated.

@jwasinger I don't fully get why you need to populate with zeroes when the same slots are overwritten by `loadMemory`?

Note, `loadMemory` should write to `returnValue[0 ... size]` and read from `memory[offset .. offset+size]`.